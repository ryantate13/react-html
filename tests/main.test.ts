import {main} from '../src';

describe('main', () => {
    it('renders with children', () => {
        expect(main({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(main({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(main({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(main()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(main({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(main({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(main({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(main({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(main({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(main({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(main({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(main({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(main({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(main({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(main({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(main({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(main({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(main({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(main({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(main({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(main({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(main({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(main({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(main({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(main({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(main({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(main({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(main({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(main({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(main({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(main({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(main({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(main({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(main({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(main({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(main({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(main({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(main({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(main({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(main({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(main({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(main({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(main({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(main({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(main({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(main({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(main({translate: "no"})).toMatchSnapshot();
    });
});