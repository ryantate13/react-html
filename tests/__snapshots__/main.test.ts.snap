// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main renders arrays of children 1`] = `
<main>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</main>
`;

exports[`main renders with accessKey="a,b,c,d" 1`] = `
<main
  accessKey="a,b,c,d"
/>
`;

exports[`main renders with autoCapitalize="characters" 1`] = `
<main
  autoCapitalize="characters"
/>
`;

exports[`main renders with autoCapitalize="none" 1`] = `
<main
  autoCapitalize="none"
/>
`;

exports[`main renders with autoCapitalize="off" 1`] = `
<main
  autoCapitalize="off"
/>
`;

exports[`main renders with autoCapitalize="on" 1`] = `
<main
  autoCapitalize="on"
/>
`;

exports[`main renders with autoCapitalize="sentences" 1`] = `
<main
  autoCapitalize="sentences"
/>
`;

exports[`main renders with autoCapitalize="words" 1`] = `
<main
  autoCapitalize="words"
/>
`;

exports[`main renders with children 1`] = `
<main>
  TEST
</main>
`;

exports[`main renders with className="foo bar baz" 1`] = `
<main
  className="foo bar baz"
/>
`;

exports[`main renders with contentEditable="false" 1`] = `
<main
  contentEditable="false"
/>
`;

exports[`main renders with contentEditable="true" 1`] = `
<main
  contentEditable="true"
/>
`;

exports[`main renders with dir="auto" 1`] = `
<main
  dir="auto"
/>
`;

exports[`main renders with dir="ltr" 1`] = `
<main
  dir="ltr"
/>
`;

exports[`main renders with dir="rtl" 1`] = `
<main
  dir="rtl"
/>
`;

exports[`main renders with draggable="false" 1`] = `
<main
  draggable="false"
/>
`;

exports[`main renders with draggable="true" 1`] = `
<main
  draggable="true"
/>
`;

exports[`main renders with empty attributes 1`] = `<main />`;

exports[`main renders with hidden=false 1`] = `
<main
  hidden={false}
/>
`;

exports[`main renders with hidden=true 1`] = `
<main
  hidden={true}
/>
`;

exports[`main renders with id="TEST" 1`] = `
<main
  id="TEST"
/>
`;

exports[`main renders with inputMode="decimal" 1`] = `
<main
  inputMode="decimal"
/>
`;

exports[`main renders with inputMode="email" 1`] = `
<main
  inputMode="email"
/>
`;

exports[`main renders with inputMode="none" 1`] = `
<main
  inputMode="none"
/>
`;

exports[`main renders with inputMode="numeric" 1`] = `
<main
  inputMode="numeric"
/>
`;

exports[`main renders with inputMode="search" 1`] = `
<main
  inputMode="search"
/>
`;

exports[`main renders with inputMode="tel" 1`] = `
<main
  inputMode="tel"
/>
`;

exports[`main renders with inputMode="text" 1`] = `
<main
  inputMode="text"
/>
`;

exports[`main renders with inputMode="url" 1`] = `
<main
  inputMode="url"
/>
`;

exports[`main renders with is="foo" 1`] = `
<main
  is="foo"
/>
`;

exports[`main renders with itemID="https://google.com/" 1`] = `
<main
  itemID="https://google.com/"
/>
`;

exports[`main renders with itemProp="https://google.com/" 1`] = `
<main
  itemProp="https://google.com/"
/>
`;

exports[`main renders with itemRef="foo bar" 1`] = `
<main
  itemRef="foo bar"
/>
`;

exports[`main renders with itemScope=false 1`] = `
<main
  itemScope={false}
/>
`;

exports[`main renders with itemScope=true 1`] = `
<main
  itemScope={true}
/>
`;

exports[`main renders with itemType="https://google.com/" 1`] = `
<main
  itemType="https://google.com/"
/>
`;

exports[`main renders with lang="" 1`] = `
<main
  lang=""
/>
`;

exports[`main renders with lang="es-419" 1`] = `
<main
  lang="es-419"
/>
`;

exports[`main renders with slot="TEST" 1`] = `
<main
  slot="TEST"
/>
`;

exports[`main renders with spellCheck="false" 1`] = `
<main
  spellCheck="false"
/>
`;

exports[`main renders with spellCheck="true" 1`] = `
<main
  spellCheck="true"
/>
`;

exports[`main renders with style={"color":"#000000"} 1`] = `
<main
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`main renders with tabIndex=1 1`] = `
<main
  tabIndex={1}
/>
`;

exports[`main renders with title="TEST" 1`] = `
<main
  title="TEST"
/>
`;

exports[`main renders with translate="no" 1`] = `
<main
  translate="no"
/>
`;

exports[`main renders with translate="yes" 1`] = `
<main
  translate="yes"
/>
`;

exports[`main renders without attributes 1`] = `<main />`;
